File: FriendsManager.BBualdo/react-ui/src/services/CategoriesService.ts
import { url } from "@/config/config";
import { Category } from "@/models/Category";
import axios from "axios";

class CategoriesService {
  async GetCategories(): Promise<Category[]> {
    return (await axios.get(url + "/categories")).data as Category[];
  }
}

export default new CategoriesService();

File: FriendsManager.BBualdo/Services/CategoriesService.cs
using Contracts;
using Contracts.DTO;
using Data.Models;
using ModelExtensions;
using Repository;

namespace Services;

public class CategoriesService(IRepository<Category> categoriesRepository) : ICategoriesService
{
  private readonly IRepository<Category> _categoriesRepository = categoriesRepository;
  
  public async Task<IEnumerable<CategoryResDto>> GetCategoriesAsync()
  {
    var categories = await _categoriesRepository.GetAsync();

    return categories.Select(category => category.ToCategoryResponse()).ToList();
  }
}